File: wishlist-frontend/src/main/java/com/wishlist/ApiClient.java
package com.wishlist;

import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.net.http.HttpRequest.BodyPublishers;

public class ApiClient {
    private static final String BASE_URL = "http://54.234.92.98:8000";
    private static String authToken = null;
    private static HttpClient client = HttpClient.newHttpClient();

    public static void setToken(String token) {
        authToken = token;
    }

    public static String getToken() {
        return authToken;
    }

    public static boolean isLoggedIn() {
        return authToken != null;
    }

    public static void logout() {
        authToken = null;
    }

    // POST request without auth
    public static String post(String endpoint, String jsonBody) {
        try {
            HttpRequest request = HttpRequest.newBuilder()
                .uri(URI.create(BASE_URL + endpoint))
                .header("Content-Type", "application/json")
                .POST(BodyPublishers.ofString(jsonBody))
                .build();
            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
            return response.body();
        } catch (Exception e) {
            return "{\"error\": \"" + e.getMessage() + "\"}";
        }
    }

    // POST request with auth token
    public static String postAuth(String endpoint, String jsonBody) {
        try {
            HttpRequest request = HttpRequest.newBuilder()
                .uri(URI.create(BASE_URL + endpoint))
                .header("Content-Type", "application/json")
                .header("authorization", "Bearer " + authToken)
                .POST(BodyPublishers.ofString(jsonBody))
                .build();
            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
            return response.body();
        } catch (Exception e) {
            return "{\"error\": \"" + e.getMessage() + "\"}";
        }
    }

    // GET request with auth token
    public static String getAuth(String endpoint) {
        try {
            HttpRequest request = HttpRequest.newBuilder()
                .uri(URI.create(BASE_URL + endpoint))
                .header("authorization", "Bearer " + authToken)
                .GET()
                .build();
            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
            return response.body();
        } catch (Exception e) {
            return "{\"error\": \"" + e.getMessage() + "\"}";
        }
    }

    // DELETE request with auth token
    public static String deleteAuth(String endpoint) {
        try {
            HttpRequest request = HttpRequest.newBuilder()
                .uri(URI.create(BASE_URL + endpoint))
                .header("authorization", "Bearer " + authToken)
                .DELETE()
                .build();
            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
            return response.body();
        } catch (Exception e) {
            return "{\"error\": \"" + e.getMessage() + "\"}";
        }
    }
}

File: wishlist-mobile/components/ApiClient.js
import AsyncStorage from '@react-native-async-storage/async-storage';
import axios from 'axios';

const BASE_URL = 'http://54.234.92.98:8000';

const ApiClient = {
  async getToken() {
    return await AsyncStorage.getItem('token');
  },

  async setToken(token) {
    await AsyncStorage.setItem('token', token);
  },

  async removeToken() {
    await AsyncStorage.removeItem('token');
  },

  async post(endpoint, data) {
    try {
      const response = await axios.post(`${BASE_URL}${endpoint}`, data);
      return response.data;
    } catch (error) {
      return error.response?.data || { error: 'Connection error' };
    }
  },

  async postAuth(endpoint, data) {
    try {
      const token = await this.getToken();
      const response = await axios.post(`${BASE_URL}${endpoint}`, data, {
        headers: { authorization: `Bearer ${token}` }
      });
      return response.data;
    } catch (error) {
      return error.response?.data || { error: 'Connection error' };
    }
  },

  async getAuth(endpoint) {
    try {
      const token = await this.getToken();
      const response = await axios.get(`${BASE_URL}${endpoint}`, {
        headers: { authorization: `Bearer ${token}` }
      });
      return response.data;
    } catch (error) {
      return error.response?.data || { error: 'Connection error' };
    }
  },

  async deleteAuth(endpoint) {
    try {
      const token = await this.getToken();
      const response = await axios.delete(`${BASE_URL}${endpoint}`, {
        headers: { authorization: `Bearer ${token}` }
      });
      return response.data;
    } catch (error) {
      return error.response?.data || { error: 'Connection error' };
    }
  },

  async putAuth(endpoint, data) {
    try {
      const token = await this.getToken();
      const response = await axios.put(`${BASE_URL}${endpoint}`, data, {
        headers: { authorization: `Bearer ${token}` }
      });
      return response.data;
    } catch (error) {
      return error.response?.data || { error: 'Connection error' };
    }
  },
};

export default ApiClient;